src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/xattention.cpp

```diff
@@ -0,0 +1,445 @@
+            if (q_end >= B) {
+                m_cnt--;
+            }
+
+            if (m_cnt > 0) {
+                auto* q_ptr = query.ptr_v(stride - 1 - i + m_start * stride, h, 0, 0);
+                auto cur_k_block_num = m_blk + 1;
+                for (size_t n_blk = 0; n_blk < cur_k_block_num; n_blk++) {
+                    auto n_start = n_blk * S;
+                    auto* k_ptr = key_repack.ptr_v(h / groups, 0, n_start, 0);
+                    auto* c_ptr = &attn_sum_temp.at<float>({h, 0, m_start, n_blk * n_block_size});
+                    kernels[m_cnt - 1]->executeGemm(m_cnt < m_block_size,
+                                                    q_ptr,
+                                                    k_ptr,
+                                                    c_ptr,
+                                                    nullptr,
+                                                    nullptr,
+                                                    wsp.data() + wsp_size * ithr,
+                                                    scratch_a.data() + scratch_a_size * ithr);
+                }
+            }
+        });
+    }
+
+    parallel_for3d(q_num_strided, H, L, [&](size_t b, size_t h, size_t l) {
+        auto* data = attn_sum_temp.ptr<float>(h, l, b, 0);
+        auto ncausal = b + 1;
+        auto scale = 1.0 / sqrt(S) / stride / norm;
+        attn_softmax_kernel<float>(data,
+                                   reinterpret_cast<float*>(data),
+                                   scale,
+                                   nullptr,
+                                   nullptr,
+                                   nullptr,
+                                   false,
+                                   ncausal,
+                                   k_num_strided,
+                                   ov::element::f32,
+                                   ov::element::f32,
+                                   0);
+    });
+
+    PlainTensor attn_sum;
+    attn_sum.resize({H, L, q_num_blocks, k_num_blocks}, attn_sum_temp.m_element_size, attn_sum_temp.m_dt);
+    size_t src_stride = attn_sum_temp.size(3);
+    size_t dst_stride = attn_sum.size(3);
+    size_t row_num = attn_sum_temp.size(2);
+    parallel_for2d(H, L, [&](size_t h, size_t l) {
+        auto* src = attn_sum_temp.ptr<float>(h, l, 0, 0);
+        auto* dst = attn_sum.ptr<float>(h, l, 0, 0);
+#if defined(HAVE_AVX512F) || defined(HAVE_AVX2)
+        if (num_per_block == 8) {
+            sum_blocks8x8(src, row_num, src_stride, dst, dst_stride);
+        } else {
+            ref::sum_blocks_ref(src, row_num, src_stride, dst, dst_stride, num_per_block);
+        }
+#else
+        ref::sum_blocks_ref(src, row_num, src_stride, dst, dst_stride, num_per_block);
+#endif
+    });
+
+    // Find blocks
+    PlainTensor mask;
+    mask.resize({H, q_num_blocks, k_num_blocks}, sizeof(bool), ov::element::Type_t::boolean);
```

Comment:
The mask here is a temporary variable, it will be created and destroyed for each PagedAttention node, however it actually could be allocated once and shared among different nodes.